Implemented the first version of the Spaced Repetition Algorithm.

ImanYZ opened the pull request.

presnick commented on cron/crontab:
```diff
@@ -0,0 +1 @@
+#crontab
```
What is this file for?

ImanYZ replied:
I have not added this. Maybe it's added by git add.

presnick commented on inst:
```diff
@@ -0,0 +1,11 @@
+_get_toc_and_questions
```
What is this file for?

ImanYZ replied:
I have not added this. Maybe it's added by git add.

presnick commented on views/practice/spacedpractice.html:
```diff
@@ -0,0 +1,140 @@
+{{extend 'admin/instructors.html'}}
```
I don't see a controller corresponding to this. Did I miss it?

ImanYZ replied:
No. I had actually created two controllers for spaced repetition and self-regulated learning. Then you decided to remove the self-regulated and I removed the controller, but forgot to remove this file.

presnick replied:
It looks like this file is still in the repo...

presnick commented:
Please remove all these unneeded files.

You can do that with `git rm <fname>`

presnick commented on controllers/assignments.py:
```diff
@@ -660,8 +695,35 @@ def _autograde_one_q(course_name, sid, question_name, points, question_type, dea
+                         (db.useinfo.timestamp >= practice_start_time)) \
+            .select()
+        practice_duration = (datetime.datetime.now() - practice_start_time).seconds / 60
+        practice_score = 0
```
Please remove the print statements or comment them out.

ImanYZ replied:
In my last commit, I had already removed all the print statements. There are only two of them remaining that are yours.

presnick commented on views/practice/index.html:
```diff
@@ -0,0 +1,139 @@
+{{extend 'admin/instructors.html'}}
```
Is this file now moved to views/admin/practice/index.html? If so, please remove this from the git repo with `git rm`

ImanYZ commented on controllers/assignments.py:
```diff
@@ -590,37 +621,41 @@ def _autograde_one_q(course_name, sid, question_name, points, question_type, dea
         else:
             event_filter = None
-        results = _scorable_useinfos(course_name, sid, question_name, points, deadline, event_filter)
+        results = _scorable_useinfos(course_name, sid, question_name, points, deadline, event_filter, None,
```
Done.

ImanYZ commented on controllers/assignments.py:
```diff
@@ -590,37 +621,41 @@ def _autograde_one_q(course_name, sid, question_name, points, question_type, dea
     elif question_type == 'page':
         # question_name does not help us
-        results = _scorable_useinfos(course_name, sid, question_name, points, deadline, question_type='page')
+        results = _scorable_useinfos(course_name, sid, question_name, points, deadline, None, 'page',
```
Done.

ImanYZ commented:
I removed all the unnecessary files.

--
Iman YeckehZaare
Ph.D. Candidate
University of Michigan School of Information

[image: http://www.iman.one/] <http://www.iman.one/>   [image:
https://www.linkedin.com/in/oneweb/] <https://www.linkedin.com/in/oneweb/>
 [image: https://stackoverflow.com/users/2521204/iman]
<https://stackoverflow.com/users/2521204/iman>   [image:
https://en.wikipedia.org/wiki/User:I.yeckehzaare]
<https://en.wikipedia.org/wiki/User:I.yeckehzaare>   [image:
https://www.si.umich.edu/people/iman-yeckehzaare]
<https://www.si.umich.edu/people/iman-yeckehzaare>

On Wed, Dec 27, 2017 at 7:39 PM, Paul Resnick <notifications@github.com>
wrote:

> *@presnick* commented on this pull request.
> ------------------------------
>
> In controllers/assignments.py
> <https://github.com/presnick/RunestoneServer/pull/166#discussion_r158859319>
> :
>
> > +        print ("practice_duration: ", practice_duration)
> +        print ("len(results): ", len(results))
> +        print ("score: ", score)
> +        print ("points: ", points)
> +        if score == points:
> +            if len(page_visits) <= 1 and len(results) <= 1 and practice_duration <= 2:
> +                practice_score = 5
> +            elif len(results) <= 2 and practice_duration <= 2:
> +                practice_score = 4
> +            elif len(results) <= 3 and practice_duration <= 3:
> +                practice_score = 3
> +            elif len(results) <= 4 and practice_duration <= 4:
> +                practice_score = 2
> +            elif len(results) <= 5 and practice_duration <= 5:
> +                practice_score = 1
> +        print ("practice_score = ", practice_score)
>
> please remove this print statement, too.
> ------------------------------
>
> In controllers/assignments.py
> <https://github.com/presnick/RunestoneServer/pull/166#discussion_r158859754>
> :
>
> > @@ -590,37 +621,41 @@ def _autograde_one_q(course_name, sid, question_name, points, question_type, dea
>              event_filter = 'unittest'
>          else:
>              event_filter = None
> -        results = _scorable_useinfos(course_name, sid, question_name, points, deadline, event_filter)
> +        results = _scorable_useinfos(course_name, sid, question_name, points, deadline, event_filter, None,
>
> use keyword-based parameter passing in order to skip the question_type
> parameter, rather than explicitly filling in None.
> results = _scorable_useinfos(course_name, sid, question_name, points,
> deadline, event_filter, practice_start_time=practice_start_time)
> ------------------------------
>
> In controllers/assignments.py
> <https://github.com/presnick/RunestoneServer/pull/166#discussion_r158859999>
> :
>
> >          scoring_fn = _score_one_mchoice
>      elif question_type == 'page':
>          # question_name does not help us
> -        results = _scorable_useinfos(course_name, sid, question_name, points, deadline, question_type='page')
> +        results = _scorable_useinfos(course_name, sid, question_name, points, deadline, None, 'page',
>
> same thing here: use keyword-based parameter passing:
> results = _scorable_useinfos(course_name, sid, question_name, points,
> deadline, question_type='page', practice_start_time=practice_start_time)
> ------------------------------
>
> In views/practice/index.html
> <https://github.com/presnick/RunestoneServer/pull/166#discussion_r158883539>
> :
>
> > @@ -0,0 +1,139 @@
> +{{extend 'admin/instructors.html'}}
>
> Is this file now moved to views/admin/practice/index.html? If so, please
> remove this from the git repo with git rm
> ------------------------------
>
> In views/practice/spacedpractice.html
> <https://github.com/presnick/RunestoneServer/pull/166#discussion_r158883558>
> :
>
> > @@ -0,0 +1,140 @@
> +{{extend 'admin/instructors.html'}}
>
> It looks like this file is still in the repo...
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/presnick/RunestoneServer/pull/166#pullrequestreview-85724008>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADqClWEUL208AInoW-MFU2nZpF3G8qdSks5tEuMvgaJpZM4RNBvi>
> .
>

presnick commented:
This has been merged to branch feature_practice.
